app/services/aws/credentials_service.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Aws
+  class CredentialsService
+    class << self
+      def call(params = {})
+        new(params).call
+      end
+    end
+
+    attr_reader :params
+
+    ROLE_SESSION_NAME = '3scale-porta'
+
+    def initialize(params)
+      @params = params
+    end
+
+    # @return [Hash] Depending on the available connections, this method may return:
+    # { credentials: Aws::STS::Types::AssumeRoleWithWebIdentityResponse } if STS credentials are available;
+    # { web_identity_token_file: 'foo', role_name: 'bar' } if the values are present;
```

Comment:
I read is as if  `{ web_identity_token_file: 'foo', role_name: 'bar' }` can be a return value. While from code it seems like that if `{ web_identity_token_file: 'foo', role_name: 'bar' }` is present, then return value is `{ credentials: Aws::STS::Types::AssumeRoleWithWebIdentityResponse }`.

May need some rewording.

Reply:
btw I expected that we will check for the token file at a predefined location instead of requiring configuration. I guess it depends how operator implements this.

Reply:
wrt `role_name`, is this something that administrator assigns to us or is it something that we always need? I mean if we need a specific role that is global to any S3 account, maybe we should hardcode the proper value. I don't know how it works so excuse my ignorance.

Reply:
>btw I expected that we will check for the token file at a predefined location instead of requiring configuration. I guess it depends how operator implements this.

The file should be on a predefined location and we should know its location through configuration. As explained in [here](https://github.com/openshift/cloud-credential-operator/blob/master/docs/sts.md), we should have access to both `web_identity_token_file` and `role_name` from OpenShift.

>wrt role_name, is this something that administrator assigns to us or is it something that we always need? I mean if we need a specific role that is global to any S3 account, maybe we should hardcode the proper value.

Good question! `role_name` is the ARN that should be provided by OpenShift.